### PR TITLE
ClientHello to resolve server certs

### PR DESCRIPTION
**Pull request by jeromegn**
This should make it easier to extend the data available to `ResolvesServerCert::resolver` without future breaking changes.

Since you've already made a breaking change (but didn't release yet) to the same function, I figured it would be a good time to sneak in a refactor.

Any thoughts? I wasn't sure where to put this struct.

### CODE REVIEW

**Comment by ctz** [comment]
Thanks for working on this; it looks good. Please can we have the new `ClientHello` struct live next to the declaration of `ResolvesServerCert`?

**Comment by jeromegn** [comment]
Done. Somehow this fixed travis, but I think it's just being flaky?

File: rustls/src/server/mod.rs [hunk 1]
```diff
@@ -95,26 +96,47 @@ pub trait ProducesTickets : Send + Sync {
...
+    sigschemes: &'a [SignatureScheme],
+    alpn: Option<&'a[&'a[u8]]>,
+}
+
+impl<'a> ClientHello<'a> {
+    /// Creates a new ClientHello
+    pub fn new(server_name: Option<webpki::DNSNameRef<'a>>, sigschemes:  &'a [SignatureScheme],
+    alpn: Option<&'a[&'a[u8]]>)->Self {
+        ClientHello {server_name, sigschemes, alpn}
+    }
```
Comment (Ralith) [inline rustls/src/server/mod.rs]: If the goal is to ensure forwards-compatibility, perhaps this shouldn't be public?

Comment (ctz) [reply rustls/src/server/mod.rs]: Good idea.

**Comment by ctz** [comment]
> travis, but I think it's just being flaky

Yes, travis is regularly flaky :( Especially the macos machines.

**Comment by jeromegn** [comment]
Good call re: `new` visibility.

Thanks for the quick merge!

**Pull request merged**